build-aux/snap/snapcraft.yaml

```
@@ -29,9 +28,36 @@ passthrough:
 # https://forum.snapcraft.io/t/5547/10
 parts:
   snapd:
-    # FIXME: this should probably go upstream
-    plugin: x-builddeb
+    plugin: nil
     source: .
+    build-snaps: [go/1.10/stable]
+    override-pull: |
+      snapcraftctl pull
+      # install build dependencies
+      export DEBIAN_FRONTEND=noninteractive
+      export DEBCONF_NONINTERACTIVE_SEEN=true
+      sudo apt-get build-dep -y ./
+      ./get-deps.sh --skip-unused-check
+      # set version after installing dependencies so we have all the tools here
+      snapcraftctl set-version $(./mkversion.sh --output-only)
```

```suggestion
      snapcraftctl set-version "$(./mkversion.sh --output-only)"
```

Happy to see follow ups or if you are feeling lucky, commit this suggestion and wait for green.